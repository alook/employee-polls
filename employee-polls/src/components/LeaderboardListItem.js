import * as React from "react";
import {Fragment} from "react";
import ListItem from '@mui/material/ListItem';
import ListItemText from '@mui/material/ListItemText';
import ListItemAvatar from '@mui/material/ListItemAvatar';
import Avatar from '@mui/material/Avatar';
import Divider from '@mui/material/Divider';

const LeaderboardListItem = (props) => {
  const {leaderboardUser} = props;
  return (
      <Fragment>
        <ListItem alignItems="flex-start">
          <ListItemAvatar>
            <Avatar alt={leaderboardUser.id} src={leaderboardUser.avatarURL}/>
          </ListItemAvatar>
          <ListItemText
              primary={leaderboardUser.name + " (" + leaderboardUser.id + ")"}
              secondary={"Asked: " + leaderboardUser.numQuestions + ", Answered: " + leaderboardUser.numAnswers + ", Total: " + leaderboardUser.numTotal}
          />
        </ListItem>
        <Divider variant="inset" component="li"/>
      </Fragment>
  )
}

export default LeaderboardListItem;