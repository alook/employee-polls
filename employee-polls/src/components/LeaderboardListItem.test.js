import { render } from '@testing-library/react'
import React from "react";
import LeaderboardListItem from "./LeaderboardListItem";

describe('LeaderboardListItem', () => {
  it('will match snapshot', async () => {
    const view = await render(<LeaderboardListItem leaderboardUser={{
      id: "userId",
      name: "username",
      avatarURL: "/face1.png",
      numQuestions: 2,
      numAnswers: 3,
      numTotal: 5
    }} />);
    expect(view).toMatchSnapshot()
  })
})