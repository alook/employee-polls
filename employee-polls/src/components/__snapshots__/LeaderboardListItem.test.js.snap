// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeaderboardListItem will match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-alignItemsFlexStart css-hhvo2v-MuiListItem-root"
      >
        <div
          class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart css-1m6607n-MuiListItemAvatar-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
          >
            <img
              alt="userId"
              class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
              src="/face1.png"
            />
          </div>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            username (userId)
          </span>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
          >
            Asked: 2, Answered: 3, Total: 5
          </p>
        </div>
      </li>
      <li
        class="MuiDivider-root MuiDivider-inset css-uls1yk-MuiDivider-root"
        role="separator"
      />
    </div>
  </body>,
  "container": <div>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-alignItemsFlexStart css-hhvo2v-MuiListItem-root"
    >
      <div
        class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart css-1m6607n-MuiListItemAvatar-root"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
        >
          <img
            alt="userId"
            class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
            src="/face1.png"
          />
        </div>
      </div>
      <div
        class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
        >
          username (userId)
        </span>
        <p
          class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
        >
          Asked: 2, Answered: 3, Total: 5
        </p>
      </div>
    </li>
    <li
      class="MuiDivider-root MuiDivider-inset css-uls1yk-MuiDivider-root"
      role="separator"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
